### DISCUSSION TITLE
Emulate Multiple Exposure Times using Multiple Brightness for Anycubic Machines

### DISCUSSION THREAD

**Discussion by h2ogeek** [sn4k3/UVtools#660] [Q&A] [answered]
Hi,

I read that Multiple Exposure does not play well with Anycubics, and the only way in UVtools to achieve this is to use Multiple Brightness! 

I have the following questions regarding how to setup MB and Pattern Loaded Model:

* What option shall I select if I want to test only the normal layers and not the burning layers? BTW are the Normal Layers called Based Layers, and the Burning Layers called Bottom Layers in UVtools?
* Is it possible to intrude on the variable exposure time? It is better than intruding on the brightness!

<img width="1138" alt="CleanShot 2023-02-19 at 11 42 44@2x" src="https://user-images.githubusercontent.com/36352470/219905895-a7938544-e974-42f2-a0f8-14c3dc95aeac.png">
<img width="1146" alt="CleanShot 2023-02-19 at 11 45 20@2x" src="https://user-images.githubusercontent.com/36352470/219905964-c8641851-565e-4d9d-b6be-89a557682a98.png">
<img width="945" alt="CleanShot 2023-02-19 at 11 41 42@2x" src="https://user-images.githubusercontent.com/36352470/219906133-1a3ad50c-6e94-49fe-8b53-909f68eae52c.png">

**Comment by sn4k3** [comment, marked as answer]
> What option shall I select, if I want to test only the normal layers and not the burning layers? 

By default, the burning layers are not tested, that setting come from "Exclude from" where Bottom is selected.

> BTW are the Normal Layers called Based Layers and the Burning Layers called Bottom Layers in UVtools?

In this test burning layers are the bottom layers. 
Base layers or base height is the amount of height you want for the full white block to extrude (without any details engraved), in other words is just a base. Base most of the time contains bottom and normal layers.

> Is it possible to intrude on the variable exposure time? It is better than intruding on the brightness!

You mean extrude? The stock UVtools test does extrudes on that but looks like you are using a model of your own and pattern it? That way it can only intrude, because most of the times people want to pattern complex models like the amarlabs tower or others and it can't guess the right spot to extrude the text, if using last layers it will extrude text in the air and fail. I know bottom intrudes can glue, your best bet is to manually insert a mark on top of each test and when clean the part you can write on it base with a pen the values from UVtools.

**Comment by h2ogeek** [comment]
@sn4k3 Thank you for your quick reply!

If the bottom layers are not tested in this test why the table shows a variation in the exposure time on the base layers and not in the normal layers? This model uses 28s exposure time on the burning layer = base (UVTools)

<img width="1257" alt="CleanShot 2023-02-19 at 12 48 57@2x" src="https://user-images.githubusercontent.com/36352470/219907779-926b4327-e510-4477-b244-ca1a091330ef.png">

I used the same term that USBTools use intrude. Is it possible to set a custom pattern that includes the same information that Multiple Exposure Times provides when this option `Intrude text in bottom layers with the model parameters` is enabled?

<img width="1128" alt="CleanShot 2023-02-19 at 12 51 32@2x" src="https://user-images.githubusercontent.com/36352470/219907889-0ad732fb-1b8d-4fcf-8849-ad463ddc8b64.png">

**Reply by sn4k3** [reply]
> ... why the table shows a variation in the exposure time ...

The table on that dialog always show the variations, no matter if they will be used or not.

>  same term that USBTools use intrude

Intrude mean it will cut model to make the letters, extrude is the otherwise, will add layers to create and show the letters.

> Is it possible to set a custom pattern that includes the same information that Multiple Exposure Times provides when this option Intrude text in bottom layers with the model parameters is enabled?

You can't, for that you must use the UVtools built-in test and not a pattern model that UVtools can't control the design.

**Comment by h2ogeek** [comment]
So if I choose "Exclude from" = 'bottom' -> The bottom layers (burning layers) are ignored correct? what about "bottomandbase"?

**Reply by sn4k3** [reply]
Correct. 
**BottomAndBase**, excludes the bottom layers (burning layers) + the base which you define on Object configuration (Base Height) but as you are patterning a model that setting is not used. Everything inside "Object configuration" is for UVtools stock test. So selecting bottom or BottomAndBase have same effect on your case of use.

**PS:** You are replying as answer, to correct use discussions please reply bellow answers to cascate the conversation.

**Reply by h2ogeek** [reply]
Awesome! Thank you @sn4k3 !

**Reply by h2ogeek** [reply]
I tried to decrease AA to have small Brightness increments, by using a value < 16, but it does not do that the smaller, the AA the brightness is higger!. What shall I do to reduce the brightness difference on each step?

<img width="1037" alt="CleanShot 2023-02-19 at 16 42 31@2x" src="https://user-images.githubusercontent.com/36352470/219929838-7fb889d3-b32c-4a3a-a95d-20ad818a6e5f.png">

**Reply by sn4k3** [reply]
The more AA you can use the less difference on step. So, you need to use 16x to have the smallest possible step.

**Reply by h2ogeek** [reply]
I would love to be able to have 0.10s exposure increments, but seems like that is not possible :-(

**Reply by sn4k3** [reply]
I would test with 16x first to see if printer handle it.
Then you can try a larger AA number, like 24, it might work but require testing
